hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -342,20 +345,25 @@ private Token.Builder customizeToken(
      * @param originalToken original token
      */
     private void updateTokenAttributes(
-            final TokenUpdateTransactionBody op, final Token.Builder builder, final Token originalToken) {
+            final TokenUpdateTransactionBody op,
+            final Token.Builder builder,
+            final Token originalToken,
+            final boolean isHapiCall) {
         if (op.symbol() != null && op.symbol().length() > 0) {
             builder.symbol(op.symbol());
         }
         if (op.name() != null && op.name().length() > 0) {
             builder.name(op.name());
         }
-        if (op.hasMemo()) {
+        if (op.hasMemo() && op.memo().length() > 0) {
```

Comment:
It is utterly stupid that Java distinguishes between `null` (i.e., no reference) and `empty()` (i.e., 0-length) in the case of strings.  And in a language for which one of the main selling points was (and is!) that "there are no pointers!".

It's also annoying that protoc doesn't generate methods that hide this from the user.  (And now pbj too.)

Although - take a look at the generated code in `TokenUpdateTransactionBody.Builder` - isn't it the case that the _setters_ there fix up this difference?  (That may be a new pbj innovation, I don't remember.)